src/main/java/org/openrewrite/java/security/spring/CsrfProtection.java

```diff
@@ -53,14 +53,9 @@ public Duration getEstimatedEffortPerOccurrence() {
         return Duration.ofMinutes(5);
     }
 
-    @Override
-    protected TreeVisitor<?, ExecutionContext> getApplicableTest() {
-        return new HasTypeOnClasspathSourceSet<>("org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter");
-    }
-
     @Override
     protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
-        return new UsesType<>("org.springframework.security.web.csrf.CookieCsrfTokenRepository");
+        return new HasTypeOnClasspathSourceSet<>("org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter");
```

Comment:
This change would result in this recipe running on more sources than it should. All sources from the same project will have the same `JavaSourceSet` marker listing the same classpath. So that would make this recipe applicable to all of them, even ones that don't use `WebSecurityConfigurerAdapter`. `UsesType` is the correct sort of single-source applicability test here.

```suggestion
        return new UsesType<>("org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter");
```